src/bluetooth.cpp

```diff
@@ -1708,11 +1523,13 @@ void bluetooth::deviceDiscovered(const QBluetoothDeviceInfo &device) {
                 chronoBike->deviceDiscovered(b);
                 // NOTE: Commented due to #358
                 // connect(this, SIGNAL(searchingStop()), chronoBike, SLOT(searchingStop()));
-                if (!discoveryAgent->isActive()) {
+            }
+
+            if(newDevice) {
+                if (emitSearchingStop && !discoveryAgent->isActive())
```

Comment:
In order to avoid changing behaviour I have this emitSearchingStop variable. But I wonder, am I simply refactoring a bug? Should it always emit searchingStop if the discoveryAgent isn't active?

Reply:
that signal is an inheritance of the past and i think it's quite useless nowadays. Anyway, in order to don't break things, your solution seems fine... We are always in time to remove it in the future ;)